certbot/main.py

```diff
@@ -655,6 +655,14 @@ def revoke(config, unused_plugins):  # TODO: coop with renewal config
     """Revoke a previously obtained certificate."""
     # For user-agent construction
     config.installer = config.authenticator = "None"
+
+    if config.cert_path is None and config.certname:
+        config.cert_path = storage.cert_path_for_cert_name(config, config.certname)
+    elif not config.cert_path or (config.cert_path and config.certname):
+        # intentionally not supporting --cert-path & --cert-name together,
+        # to avoid dealing with mismatched values
+        raise errors.Error("Error! Exactly one of --cert-path or --cert-name must be specified!")
```

Comment:
@ohemorange this means we can delete everything under this top conditional branch: https://github.com/certbot/certbot/blob/6f86267a26c9b748bd90113fe62157a9a455cdd2/certbot/main.py#L556, right?

When I looked at the failing integration tests, they were the put into place to stop weird behaviour as a result of users passing both `--cert-name` and `--cert-path` (e.g. https://github.com/certbot/certbot/blame/master/tests/boulder-integration.sh#L405 and https://github.com/certbot/certbot/blame/master/tests/boulder-integration.sh#L415), so I think I can delete these too now?

Reply:
oh yes, I remember that we added those tests! this all sounds right.